src/main.ts

```diff
@@ -76,23 +89,32 @@ async function getApps(): Promise<App[]> {
   } catch (e) {
     core.error(e);
   }
+
   return (responseJson.items as App[]).filter(app => {
-    // TODO filter apps to only ones where they point to paths that have changed in this repo
-    return app.spec.source.repoURL.includes(
-      `${github.context.repo.owner}/${github.context.repo.repo}`
+    return (
+      app.spec.source.repoURL.includes(
+        `${github.context.repo.owner}/${github.context.repo.repo}`
+      ) && app.spec.source.targetRevision === 'master'
     );
   });
 }
 
 interface Diff {
-  appName: string;
+  app: App;
   diff: string;
 }
 async function postDiffComment(diffs: Diff[]): Promise<void> {
-  const output = diffs
-    .map(
-      ({ appName, diff }) => `    
-ArgoCD Diff for [\`${appName}\`](https://${ARGOCD_SERVER_URL}/applications/${appName})        
+  const { owner, repo } = github.context.repo;
+  const sha = github.context.payload.pull_request?.head?.sha;
+
+  const commitLink = `https://github.com/${owner}/${repo}/pull/${github.context.issue.number}/commits/${sha}`;
+  const shortCommitSha = String(sha).substr(0, 7);
+
+  const diffOutput = diffs.map(
+    ({ app, diff }) => `    
+Diff for App: [\`${app.metadata.name}\`](https://${ARGOCD_SERVER_URL}/applications/${
+      app.metadata.name
+    }) App sync status: ${app.status.sync.status === 'Synced' ? 'Synced ✅' : 'Out of Sync ⚠️'}
```

Comment:
Could we go a step further and split the diffs into separate sections, so it's easy to scan the most relevant diffs at the top? Something like:

## 🐙 ArgoCD diffs for commit [`baaf91f`](https://github.com/quizlet/quizlet-infrastructure/pull/416/commits/baaf91feac06124f0d54f7df4d5f1223eabea7b6)

 ✅ **Up-to-date**

[`web-prod-cluster`](https://argocd.qzlt.io/applications/web-prod-cluster)
<details>

```diff
===== argoproj.io/Application argocd/webapp-prod-namespaces ======
25a26
>           - proxysql

```

</details>

---

⚠️ **Out-of-sync** 

[`web-prod-cluster`](https://argocd.qzlt.io/applications/web-prod-cluster) 
<details>

```diff
===== argoproj.io/Application argocd/webapp-prod-namespaces ======
25a26
>           - proxysql

```

</details>

Reply:
I think it'd be worth keeping the better DX (IMO) of separate sections and add notes contextualizing them for reviewers.